prog/dftb+/prg_dftb/mainio.F90

```diff
@@ -82,524 +96,1459 @@ module mainio
 
-    !> Orbital information.
-    type(TOrbitals), intent(in) :: orb
-
-    !> Species.
+    !> species of all atoms in the system
     integer, intent(in) :: species(:)
 
-    !> Name of the species.
-    character(mc), intent(in) :: speciesName(:)
+    !> label for each atomic chemical species
+    character(*), intent(in) :: speciesName(:)
 
-    !> Sparse overlap matrix.
+    !> Atomic orbital information
+    type(TOrbitals), intent(in) :: orb
+
+    !> k-points
+    real(dp), intent(in) :: kPoint(:,:)
+
+    !> sparse overlap matrix
     real(dp), intent(in) :: over(:)
 
-    !> KPoints.
-    real(dp), intent(in) :: kpoint(:,:)
+    !> K-points and spins to process
+    type(TParallelKS), intent(in) :: parallelKS
+
+    !> Whether eigenvectors should be also written in text form
+    logical, intent(in) :: tPrintEigvecsTxt
+
+    !> Complex eigenvectors (will be overwritten)
+    complex(dp), intent(inout) :: eigvecsCplx(:,:,:)
+
+    !> Storage for dense complex overlap matrix
+    complex(dp), intent(out) :: SSqrCplx(:,:)
+
+    !> optional alternative file prefix, to appear as "fileName".bin or "fileName".out
+    character(len=*), intent(in), optional :: fileName
+
+  #:if WITH_SCALAPACK
+    call writeCplxEigvecsBinBlacs(env, denseDesc, eigvecsCplx, fd, runId, parallelKS,&
+        & fileName=fileName)
+    if (tPrintEigvecsTxt) then
+      if (denseDesc%t2Component) then
+        call writePauliEigvecsTxtBlacs(env, denseDesc, eigvecsCplx, fd, parallelKS, orb, over,&
+            & kPoint, neighborList%iNeighbor, nNeighbor, iCellVec, cellVec, iPair, img2CentCell,&
+            & species, speciesName, fileName=fileName)
+      else
+        call writeCplxEigvecsTxtBlacs(env, denseDesc, eigvecsCplx, fd, parallelKS, orb, over,&
+            & kPoint, neighborList%iNeighbor, nNeighbor, iCellVec, cellVec, iPair, img2CentCell,&
+            & species, speciesName, fileName=fileName)
+      end if
+    end if
+  #:else
+    call writeCplxEigvecsBinSerial(eigvecsCplx, fd, runId, parallelKS, fileName=fileName)
+    if (tPrintEigvecsTxt) then
+      if (denseDesc%t2Component) then
+        call writePauliEigvecsTxtSerial(fd, neighborList, nNeighbor, denseDesc, iPair,&
+            & img2CentCell, iCellVec, cellVec, orb, species, speciesName, over, parallelKS, kPoint,&
+            & eigvecsCplx, SSqrCplx, fileName=fileName)
+      else
+        call writeCplxEigvecsTxtSerial(fd, neighborList, nNeighbor, denseDesc, iPair, img2CentCell,&
+            & iCellVec, cellVec, orb, species, speciesName, over, parallelKS, kPoint, eigvecsCplx,&
+            & SSqrCplx, fileName=fileName)
+      end if
+    end if
+  #:endif
+
+  end subroutine writeCplxEigvecs
+
+
+#:for DTYPE, NAME in [('complex', 'Cplx'), ('real', 'Real')]
+
+#:if WITH_SCALAPACK
+
+  !> Write the real eigvectors into binary output file (BLACS version).
+  subroutine write${NAME}$EigvecsBinBlacs(env, denseDesc, eigvecs, fd, runId, parallelKS, fileName)
```

Comment:
Might want to comment up the ${NAME} use (possibly what is killing the literate comment tools).

Reply:
Done.